### ISSUE TITLE
Using http_archive() to pull a precompiled C++ library

### ISSUE THREAD

**Issue by andyfriesen** [facebook/buck2#318]
I'm looking to write a little build rule that pulls a precompiled library off the Internet and uses it as a dependency.  I've got something like this:

```python
http_archive(
    name="archive",
    sha256="12345",
    urls=["whatever"],
    sub_targets=[
        'include',
        'lib/libwhatever.a',
    ],
)

prebuilt_cxx_library(
    name='whatever',
    visibility=['PUBLIC'],
    public_include_directories=[':archive[include]'],
    static_lib=':archive[lib/libwhatever.a]',
)
```

The problem I'm running into is that buck2 doesn't do any kind of processing on the `public_include_directories` field, so when I try to link this target, buck2 winds up passing `-Ipath/to/:archive[include]` to the compiler verbatim which obviously does not work.

Is there another way to do this?

**Comment by dtolnay** [comment]
Try this, instead of `public_include_directories`:

```starlark
prebuilt_cxx_library(
    ...
    exported_preprocessor_flags = ["-I$(location :archive[include])"],
)
```

**Comment by andyfriesen** [comment]
That did the trick.  Thanks a ton!

Is this a bug that needs fixing or is this intended?

**Comment by dtolnay** [comment]
It's intentional. I tried to fix it in 08f2a4f1131dc0beb2ba2199e65af5a5c0060f7a but it failed &mdash; see cd4d70a5f41211401db398e2fdce45d6a70c8008.

**Comment by andyfriesen** [comment]
Got it.  Thanks again!
